.github/workflows/preview.yml

```diff
@@ -4,9 +4,9 @@ name: Preview
 on: [pull_request]
 
 jobs:
-  code-engine:
+  preview-setup:
     # Prevents action to be executed by forks and drafts
-    if: github.event.pull_request.draft == false && github.event.pull_request.head.repo.full_name == github.repository
+    if: github.event.action != 'closed' && github.event.pull_request.draft == false && github.event.pull_request.head.repo.full_name == github.repository
```

Comment:
What do you think about:

1. Allow draft PR.
    - If not we need to create fake-draft PRs like [#2408](https://github.com/Qiskit/qiskit.org/pull/2408) to have previews.
2. Remove `github.event.pull_request.head.repo.full_name == github.repository`.
    - I can't imagine the use case where someone fork and creates the PR on the fork and not on the original repository.

So we just have:

```
    if: github.event.action != 'closed'
```